model_external_socket/java/src/org/openda/externalsocket/ExternalSocketModelFactory.java

```diff
@@ -0,0 +1,63 @@
+package org.openda.externalsocket;
+
+import org.openda.blackbox.interfaces.IModelFactory;
+import org.openda.blackbox.interfaces.ITimeHorizonConsumer;
+import org.openda.interfaces.IStochModelFactory;
+import org.openda.interfaces.IStochModelPostProcessor;
+import org.openda.interfaces.ITime;
+
+import java.io.File;
+
+public class ExternalSocketModelFactory implements IModelFactory, ITimeHorizonConsumer, IStochModelFactory {
+
+	private ExternalSocketModelFactoryConfigReader configReader;
+	private File dummyModelDir;
+
+	public ExternalSocketModelFactory() {
```

Comment:
Codacy found an issue: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/gh/OpenDA-Association/OpenDA/pullRequest?prid=6157987)